ngraph/python/src/ngraph/utils/node_factory.py

```diff
@@ -30,4 +33,67 @@ def create(
         if attributes is None:
             attributes = {}
         node = self.factory.create(op_type_name, arguments, attributes)
+
+        # Set getters and setters for each node's attribute.
+        #   node.get_attribute_name()
+        #   node.set_attribute_name()
+        # For compound (hierarchical) attributes of form ie.: node.struct_member_name.attr_name:
+        #   node.get_struct_member_name_attr_name()
+        #   node.set_struct_member_name_attr_name()
+        all_attributes = node._get_attributes()
+        for attr_name in all_attributes.keys():
+            setattr(node,
+                    self._normalize_attr_name_getter(attr_name),
+                    partial(NodeFactory._get_node_attr_value, node, attr_name))
+            setattr(node,
+                    self._normalize_attr_name_setter(attr_name),
+                    partial(NodeFactory._set_node_attr_value, node, attr_name))
+
+        # Setup helper members for caching attribute values.
+        setattr(node, "_attr_cache", all_attributes)
```

Comment:
What do you think about creating the cache rather with the first `get` call instead of here. 
Just to save some memory if user don't even use these attrs after the creation of the node.